### PR TITLE
[Enhancement] Support load segment parallel in shared_data

### PR DESCRIPTION
**Pull request by smartlxh**
Why I'm doing:
Query may be slow when cache miss if there are many rowset or segments, for Initialization  of segment is serial。 
What I'm doing:
Initialization of Rowsets and segments asynchronous 
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/lake/tablet_reader.cpp [hunk 1]
```diff
@@ -149,9 +150,27 @@ Status TabletReader::get_segment_iterators(const TabletReaderParams& params, std
...
     SCOPED_RAW_TIMER(&_stats.create_segment_iter_ns);
+
+    std::vector<std::future<StatusOr<std::vector<ChunkIteratorPtr>>>> futures;
+    auto load_segment_thread_pool = ExecEnv::GetInstance()->load_segment_thread_pool();
+
     for (auto& rowset : _rowsets) {
-        ASSIGN_OR_RETURN(auto seg_iters, enhance_error_prompt(rowset->read(schema(), rs_opts)));
-        iters->insert(iters->end(), seg_iters.begin(), seg_iters.end());
+        auto task = std::make_shared<std::packaged_task<StatusOr<std::vector<ChunkIteratorPtr>>()>>(
+                [&, rowset]() { return enhance_error_prompt(rowset->read(schema(), rs_opts)); });
```
[the remark this answers is not in the document]
Comment (smartlxh) [reply be/src/storage/lake/tablet_reader.cpp:159]: I think it's ok, for the situation exist when there are concurrent querys?  PTAL @sduzh

File: be/src/storage/lake/rowset.cpp [hunk 2]
```diff
@@ -233,14 +235,36 @@ Status Rowset::load_segments(std::vector<SegmentPtr>* segments, bool fill_data_c
...
+        segment_futures.push_back(task->get_future());
+    }
+
+    for (auto& fut : segment_futures) {
+        auto segment_or = fut.get();
         if (segment_or.ok()) {
             segments->emplace_back(std::move(segment_or.value()));
         } else if (segment_or.status().is_not_found() && ignore_lost_segment) {
-            LOG(WARNING) << "Ignored lost segment " << seg_name;
+            LOG(WARNING) << "Ignored lost segment";
```
Comment (starrocks-xupeng) [inline be/src/storage/lake/rowset.cpp:267]: should log file name here, otherwise the log doesn't help anything

File: be/src/common/config.h [hunk 3]
```diff
@@ -417,6 +417,9 @@ CONF_Int32(make_snapshot_rpc_timeout_ms, "20000");
 // OlapTableSink sender's send interval, should be less than the real response time of a tablet writer rpc.
 CONF_mInt32(olap_table_sink_send_interval_ms, "10");
 
+CONF_Int32(load_segment_thread_pool_num_max, "128");
+CONF_Int32(load_segment_thread_pool_queue_size, "10240");
```
Comment (starrocks-xupeng) [inline be/src/common/config.h:421]: should add an option to control whether enable parallel fetching? in case anything bad happens

Comment (smartlxh) [reply be/src/common/config.h:422]: done

File: be/src/storage/lake/rowset.cpp (hunk 2 above)
Comment (smartlxh) [reply be/src/storage/lake/rowset.cpp]: done

File: be/src/runtime/exec_env.h [hunk 4]
```diff
@@ -339,6 +339,8 @@ class ExecEnv {
     PriorityThreadPool* _thread_pool = nullptr;
     ThreadPool* _streaming_load_thread_pool = nullptr;
 
+    ThreadPool* _load_segment_thread_pool = nullptr;
```
[the remark this answers is not in the document]
Comment (smartlxh) [reply be/src/runtime/exec_env.h:344]: done

**Comment by tracymacding** [comment]
@mergify backport branch-3.2-jeff-io

**Comment by yingtingdong** [comment]
ignore backport check: 3.2.7

**Comment by evelynzhaojie** [comment]
@smartlxh are these parameters visible to users, if yes, would you pls add the parameter description to https://docs.starrocks.io/zh/docs/3.3/administration/management/BE_configuration/

**Comment by yingtingdong** [comment]
ignore backport check: 3.2.8

File: be/src/runtime/exec_env.cpp [hunk 5]
```diff
@@ -667,6 +687,7 @@ void ExecEnv::destroy() {
     SAFE_DELETE(_connector_scan_executor);
     SAFE_DELETE(_thread_pool);
     SAFE_DELETE(_streaming_load_thread_pool);
+    SAFE_DELETE(_load_segment_thread_pool);
```
Comment (wxl24life) [inline be/src/runtime/exec_env.cpp:690]: seem like `_load_rowset_thread_pool` is missed to delete here?

**Pull request merged**